Josh

JeonghoonWon opened the pull request:
db backup & delete milestone

JeonghoonWon committed:
GAIA_DB_backup

210529 ~ 210602 GAIA_DB_backup

JeonghoonWon committed:
delete milestone

delete milestone

JeonghoonWon committed:
Merge remote-tracking branch 'origin/master' into josh

ShanePark commented on gaia/src/main/java/best/gaia/issue/controller/MilestoneREST.java:
```diff
@@ -128,10 +129,27 @@ public void init() {
+	@DeleteMapping()
+	public Map<String, Object> deleteMilestone(
+			HttpSession session
+			,@RequestParam (required=false) Integer milest_no
```
required=  false 지우기

ShanePark commented on gaia/src/main/java/best/gaia/issue/controller/MilestoneREST.java:
```diff
@@ -128,10 +129,27 @@ public void init() {
+			, @ModelAttribute MilestoneVO search
+				) {
+		
+		System.err.println(milest_no);
```
sys err 지우기

ShanePark commented on gaia/src/main/java/best/gaia/issue/controller/MilestoneREST.java:
```diff
@@ -128,10 +129,27 @@ public void init() {
+	public Map<String, Object> deleteMilestone(
+			HttpSession session
+			,@RequestParam (required=false) Integer milest_no
+			, @ModelAttribute MilestoneVO search
```
modelAttribute 받거나 milest_no 받거나 한개만 하기

ShanePark commented on gaia/src/main/java/best/gaia/issue/controller/MilestoneREST.java:
```diff
@@ -128,10 +129,27 @@ public void init() {
-	@RequestMapping(method=RequestMethod.DELETE)
-	public Map<String, Object> deleteMilestone() {
-		return null;
+		Integer proj_no = getProjNoFromSession(session);
```
이게 세션을 받아오는 건데 milst_sid 를 통해서 삭제하면 세션을 받아 올 필요가 없음

ShanePark commented on gaia/src/main/resources/best/gaia/db/mybatis/mappers/milestone.xml:
```diff
@@ -126,7 +126,11 @@ SELECT DISTINCT ISSUE.ISSUE_SID AS ISSUE_ID, ISSUE.ISSUE_NO, ISSUE.ISSUE_TITLE,I
 	
-
+	<delete id="deleteMilestone" parameterType="best.gaia.vo.MilestoneVO">
+			 delete from milestone
+   			 where milest_no = #{milest_no,jdbcType=NUMERIC} and proj_no = #{proj_no,jdbcType=NUMERIC}
```
milst_sid 이용해서 삭제하기

ShanePark reviewed (changes requested):
milst_sid 를 parameter로 받아서 삭제하게끔 수정해주세요

JeonghoonWon committed:
milestone ...

milestone 일정글자 이상 ... 처리
new milestone 등록 시 required 경우 create button disabled 처리

JeonghoonWon committed:
Merge remote-tracking branch 'origin/master' into josh

ShanePark commented:
수정 후에 slack 으로 알려 주세요 !

JeonghoonWon committed:
form 에 submit 버튼 생성

최희연 선생님께서 form 에 submit 버튼이 없는경우, 일반 버튼이 submit 기능을 대체할 수 있다고 하셨습니다. hidden 으로 submit 버튼을 생성하면 방지할 수 있다고 합니다.
시작일과 종료일을 입력 하지 않은 경우 등록값 없음으로 뜨도록 처리중.

JeonghoonWon committed:
Merge remote-tracking branch 'origin/master' into josh

ShanePark commented on gaia/src/main/java/best/gaia/issue/service/IssueServiceImpl.java:
```diff
@@ -106,8 +106,9 @@ public ServiceResult updateMilestone(MilestoneVO milestone) {
 	public ServiceResult deleteMilestone(MilestoneVO search) {
-		// TODO Auto-generated method stub
-		return null;
+		int result = milestoneDao.deleteMilestone(search);
+		
+		return result==1? ServiceResult.OK : ServiceResult.FAIL;	
 	}
```
마일스톤 삭제하는 로직은 , 속하는 이슈가 있는지 없는지에 따라 달려져야 할 것 같아요.
삭제하고자 하는 마일스톤에 속한 이슈들이 존재하는 경우에는, 마일스톤 삭제를 못하게 할 것인지 
아니면 마일스톤 삭제시 해당하는 이슈들의 마일스톤 번호가 null 이 되게끔 할건지.
후자의 경우에는 삭제 dao 쿼리를 치기 전에, 먼저 해당 마일스톤에 속한 모든 이슈들의 마일스톤 정보를 null로 바꾸는 dao를 먼저 해줘야 해요

ShanePark reviewed (approved):
수고하셨어요! PR은 받지만 리뷰 내용 한번 고민해보세요 !
